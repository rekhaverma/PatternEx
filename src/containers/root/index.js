export { default } from './root.container.jsx';
