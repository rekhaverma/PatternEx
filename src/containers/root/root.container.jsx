import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { routerActions } from 'react-router-redux';

import { windowLocation } from 'lib/window-location';
import { checkIfUserIsLoggedIn } from 'model/actions/session';
import { defaultRoute } from 'config';
import Notification from 'containers/notification';
import Loader from 'components/loader/loader-v2.component';

export class Root extends React.PureComponent {
  constructor(...args) {
    super(...args);

    this.state = {
      sessionCheck: false,
    };
  }

  componentWillMount() {
    this.props.checkIfUserIsLoggedIn();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.sessionCheck.accessCheck !== this.props.sessionCheck.accessCheck) {
      if (nextProps.sessionCheck.isAuthenticated) {
        if (nextProps.location.pathname === '/login' || nextProps.location.pathname === '/') {
          this.props.updateLocation(defaultRoute);
        }
      } else if (nextProps.location.pathname !== '/login') {
        windowLocation.replace('/login');
      }

      this.setState({
        sessionCheck: true,
      });
    }
  }

  render() {
    return (
      <div className="root">
        {!this.state.sessionCheck && <Loader />}
        {this.state.sessionCheck && this.props.children}
        <div className="notifications-wrapper">
          {
            this.props.notifications.map((notification, i) => (
              <Notification
                key={`notification${i}`}
                id={notification.id}
                type={notification.type}
                content={notification.content}
              />
            ))
          }
        </div>
      </div>
    );
  }
}

Root.displayName = 'Root';
Root.propTypes = {
  location: PropTypes.object,
  sessionCheck: PropTypes.object.isRequired,
  children: PropTypes.element.isRequired,
  checkIfUserIsLoggedIn: PropTypes.func.isRequired,
  updateLocation: PropTypes.func.isRequired,
  notifications: PropTypes.array,
};
Root.defaultProps = {
  location: {},
  notifications: [],
};

export const mapStateToProps = state => ({
  sessionCheck: state.session.toJS().sessionCheck,
  notifications: state.app.ui.toJS().notifications,
});

const mapDispatchToProps = dispatch => ({
  updateLocation: location => dispatch(routerActions.push(location)),
  checkIfUserIsLoggedIn: (...args) => dispatch(checkIfUserIsLoggedIn(...args)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Root);
