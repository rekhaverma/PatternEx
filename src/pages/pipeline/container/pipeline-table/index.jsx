export { default } from './pipeline-table.container';
