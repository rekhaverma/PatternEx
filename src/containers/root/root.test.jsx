import React from 'react';
import sinon from 'sinon';
import { fromJS } from 'immutable';

import { Root, mapStateToProps } from './root.container';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<Root />', () => {
  describe('render: ', () => {
    it('should match with snapshot - loader enabled', () => {
      const props = {
        location: {},
        sessionCheck: {},
        notifications: [],
        checkIfUserIsLoggedIn: jest.fn(),
        updateLocation: jest.fn(),
      };
      const wrapper = mount(<Root {...props} children={<div />} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should match with snapshot - loader disabled', () => {
      const props = {
        location: {},
        sessionCheck: {},
        notifications: [],
        checkIfUserIsLoggedIn: jest.fn(),
        updateLocation: jest.fn(),
      };
      const wrapper = mount(<Root {...props} children={<div />} />);
      wrapper.instance().componentWillReceiveProps({
        ...props,
        sessionCheck: {
          accessCheck: true,
          isAuthenticated: true,
        },
      });
      expect(wrapper.update()).toMatchSnapshot();
    });
  });

  describe('mapStateToProps: ', () => {
    it('should return states', () => {
      const initialState = {
        session: fromJS({
          sessionCheck: {
            accessCheck: true,
            isAuthenticated: true,
          },
        }),
        app: {
          ui: fromJS({
            notifications: [],
          }),
        },
      };
      const expectedResponse = {
        notifications: [],
        sessionCheck: {
          accessCheck: true,
          isAuthenticated: true,
        },
      };
      expect(mapStateToProps(initialState)).toEqual(expectedResponse);
    });
  });

  describe('actions: ', () => {
    it('should dispatch "checkIfUserIsLoggedIn" on componentWillMount lifecycle', () => {
      const checkIfUserIsLoggedInSpy = sinon.spy();
      const props = {
        location: {},
        sessionCheck: {},
        notifications: [],
        checkIfUserIsLoggedIn: checkIfUserIsLoggedInSpy,
        updateLocation: jest.fn(),
      };
      shallow(<Root {...props} children={<div />}/>);

      expect(checkIfUserIsLoggedInSpy.callCount).toEqual(1);
    });
  });
});
