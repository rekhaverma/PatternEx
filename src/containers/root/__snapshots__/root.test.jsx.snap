// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> render:  should match with snapshot - loader disabled 1`] = `
<Root
  checkIfUserIsLoggedIn={[Function]}
  location={Object {}}
  notifications={Array []}
  sessionCheck={Object {}}
  updateLocation={[Function]}
>
  <div
    className="root"
  >
    <div />
    <div
      className="notifications-wrapper"
    />
  </div>
</Root>
`;

exports[`<Root /> render:  should match with snapshot - loader enabled 1`] = `
<Root
  checkIfUserIsLoggedIn={[Function]}
  location={Object {}}
  notifications={Array []}
  sessionCheck={Object {}}
  updateLocation={[Function]}
>
  <div
    className="root"
  >
    <Loader
      small={false}
    >
      <div
        className="loaderV2"
      >
        <div
          className="loaderV2__content"
        >
          <div
            className="rowTriangle"
          >
            <div
              className="arrow up outer outer-18"
            />
            <div
              className="arrow down outer outer-17"
            />
            <div
              className="arrow up outer outer-16"
            />
            <div
              className="arrow down outer outer-15"
            />
            <div
              className="arrow up outer outer-14"
            />
          </div>
          <div
            className="rowTriangle"
          >
            <div
              className="arrow up outer outer-1"
            />
            <div
              className="arrow down outer outer-2"
            />
            <div
              className="arrow up inner inner-6"
            />
            <div
              className="arrow down inner inner-5"
            />
            <div
              className="arrow up inner inner-4"
            />
            <div
              className="arrow down outer outer-13"
            />
            <div
              className="arrow up outer outer-12"
            />
          </div>
          <div
            className="rowTriangle"
          >
            <div
              className="arrow down outer outer-3"
            />
            <div
              className="arrow up outer outer-4"
            />
            <div
              className="arrow down inner inner-1"
            />
            <div
              className="arrow up inner inner-2"
            />
            <div
              className="arrow down inner inner-3"
            />
            <div
              className="arrow up outer outer-11"
            />
            <div
              className="arrow down outer outer-10"
            />
          </div>
          <div
            className="rowTriangle"
          >
            <div
              className="arrow down outer outer-5"
            />
            <div
              className="arrow up outer outer-6"
            />
            <div
              className="arrow down outer outer-7"
            />
            <div
              className="arrow up outer outer-8"
            />
            <div
              className="arrow down outer outer-9"
            />
          </div>
        </div>
      </div>
    </Loader>
    <div
      className="notifications-wrapper"
    />
  </div>
</Root>
`;
